libfreerdp/core/license.c

```diff
@@ -491,11 +487,16 @@ static BOOL computeCalHash(const char* hostname, char* hashStr)
 	if (!winpr_Digest_Final(sha1, hash, sizeof(hash)))
 		goto out;
 
-	for (i = 0; i < sizeof(hash); i++, hashStr += 2)
+	if (len < 2 * sizeof(hash) + 1)
+		return FALSE;
```

Comment:
Should be `goto out;` or we leak `sha1`.

Reply:
fixed. good catch.